src/lsp/cobol_ptree/compilation_group.ml

```diff
@@ -28,7 +28,7 @@ type program_unit =
     program_env: environment_division with_loc option;
     program_data: data_division with_loc option;
     program_proc: procedure_division with_loc option;
-    program_end_name: name with_loc option
+    program_end_name: ident_or_nonnum with_loc option
```

Comment:
I think this should be `name_or_alphanum` instead.

Reply:
I have tried, but I gave up. I cannot even understand how to translate an `ident_or_aphanum` into a `string` in `terms.ml`.

Reply:
What do you mean? Here we need `name_or_alphanum`, not `ident_or_alphanum`. Why do you need a `string`?

Reply:
Because later in the analysis phases, some functions want to get a string for the program name, so I needed to do some conversions. But the Terms format and how to access it remains a complete mystery for me...